libnfc/nfc.c

```diff
@@ -587,8 +588,6 @@ nfc_initiator_select_passive_target(nfc_device *pnd,
     szInit = szInitData;
   }
   HAL(initiator_select_passive_target, pnd, nm, abtInit, szInit, pnt);
-
-  free(abtTmpInit);
```

Comment:
What are you trying to fix?  A double-free error?  If so, I feel like this should be the only free() after `nfc_device_validate_modulation()` returns successfuly.